components/_Progress-Bar/progress-bar.js

```diff
@@ -0,0 +1,63 @@
+const msdsProgressBar = {
+  //Get the base number for the progress bar
+  parentElement: '',
+  baseNumber: 0,
+  currentStep: 0,
+  startFromOne: false,
+  init: function(parentElementId, baseNumber, customProgressionStep) {
+    this.parentElement = document.getElementById(parentElementId)
+
+    if (customProgressionStep > 0) {
+      this.currentStep = customProgressionStep
+    } else if (this.startFromOne == true) {
+      this.currentStep = 1
+    } else {
+      this.currentStep = 0
+    }
+    this.baseNumber = baseNumber
+    this.updateProgressbarDisplay()
+  },
+  updateProgressbarDisplay: function() { //update the display of the prorgess bar (step and counter)
+    
+    const progressBarProgressionElement = this.parentElement.querySelector('.msds-progress-bar__progression-bar')
+    if (progressBarProgressionElement) {
+      //calculate the percentage width of the progress step and update the look and fill base on this
+      const updatedWidth = (this.currentStep * 100) / this.baseNumber
+      progressBarProgressionElement.style.width = updatedWidth + '%'
+      
+      //Add the completed class to round the corners of the progression bar.
+      if (this.currentStep == this.baseNumber) {
+        progressBarProgressionElement.classList.add('msds-progress-bar__progression-bar--completed')
```

Comment:
'msds-progress-bar__progression-bar--completed' string used three times? Maybe should be moved to constants on top of object definition? I am usually moving all selectors to constans or even better move entire dom element to variable in initialization so browser doesn't need to select it from dom every time when function is executed.